salt/config.py

```diff
@@ -1411,7 +1412,10 @@ def _read_conf_file(path):
             conf_opts = {}
         # allow using numeric ids: convert int to string
         if 'id' in conf_opts:
-            conf_opts['id'] = str(conf_opts['id'])
+            if not isinstance(conf_opts['id'], six.string_types):
+                conf_opts['id'] = str(conf_opts['id'])
```

Comment:
The reason for this broken logic is that for a reason that I'm unaware of, sdecode is a noop if its argument is not string_types.